library/Vanilla/Embeds/TwitchEmbed.php

```diff
@@ -0,0 +1,209 @@
+<?php
+/**
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license https://opensource.org/licenses/GPL-2.0 GPL-2.0
+ */
+
+namespace Vanilla\Embeds;
+
+use Exception;
+
+/**
+ * Class for parsing twitch URLs, fetching twitch OEmbed data, and rendering server side HTML for a twitch video embed
+ */
+class TwitchEmbed extends VideoEmbed {
+
+    const DEFAULT_HEIGHT = 300;
+
+    const DEFAULT_WIDTH = 400;
+
+    /** @inheritdoc */
+    protected $domains = ['www.twitch.tv', 'clips.twitch.tv'];
+
+    /** @var string */
+    private $urlType;
```

Comment:
Why would you want to alter the state of a parser to record an attribute of an item it's parsed? The class is for parsing Twitch URLs, but now it has a "URL type" that doesn't seem to fit the purpose. Instead, could you have implemented a method to get the "type" from a URL and passed its return value to `getEmbedUrl`? It looks like you're trying to use `getUrlInformation` to parse out two bits of information, but you're only returning one and "saving" the other value in the state of the object. Because of this, you're relying on things to run in a specific order when they don't have to.